code/modules/integrated_electronics/core/assemblies.dm

```diff
@@ -326,8 +317,13 @@
 		interact(user)
 		return TRUE
 	else
+		if(user.a_intent == INTENT_HELP)
+			for(var/obj/item/integrated_circuit/input/objscaner/S in assembly_components)
```

Comment:
end the type path at /input
this should be able to support other kinds of input circuits if need be

Reply:
and have it work with all intents while you're at it, just pass the intent in as an argument